ape_ens/converters.py

```diff
@@ -1,49 +1,57 @@
+from pathlib import Path
 from typing import Any
 
-from ape.api import ConverterAPI, Web3Provider
-from ape.exceptions import ProviderError
-from ape.logging import logger
+from ape.api import ConfigDict, ConfigItem, ConverterAPI
 from ape.types import AddressType
 from ape.utils import cached_property
-from web3.main import ENS
+from ethpm_types import PackageManifest
 
+from .utils import get_node_id
 
-class ENSConversions(ConverterAPI):
-    """Converts ENS names like `my-name.eth` to `0xAbCd...1234`"""
-
-    @cached_property
-    def ens(self) -> ENS:
-        provider = self.networks.active_provider
+# TODO: Store/Load from IPFS
+ENS_PACKAGE = PackageManifest.parse_file(Path(__file__).parent / "ens-manifest.json")
 
-        if not provider:
-            raise ProviderError("Not connected to a provider.")
 
-        if not isinstance(provider, Web3Provider):
-            raise NotImplementedError("Currently, only web3 providers work with this plugin.")
+class ENSConfig(ConfigItem):
+    provider_name: str = "geth"
+    provider_settings: ConfigDict = {}  # type: ignore
 
-        web3 = provider._web3
 
-        if not hasattr(web3, "ens"):
-            raise NotImplementedError("This provider does not implement ENS calls.")
+class ENSConversions(ConverterAPI):
+    """Converts ENS names like `my-name.eth` to `0xAbCd...1234`"""
 
-        return web3.ens
+    @cached_property
+    def registrar(self):
+        from ape import Contract
+
+        # NOTE: ENS Only references data from Ethereum Mainnet
+        registrar_deployment = list(ENS_PACKAGE.deployments.values())[0]["Registrar"]
+        with self.networks.ethereum.mainnet.use_provider(
```

Comment:
If we are already connected to mainnet, we could just use the active provider and not have to make 2 connections

Reply:
Fair, although this seems like an optimization to make to the context manager